### PR TITLE
Actually use version-env on mumctl

### PR DESCRIPTION
**Pull request by sornas**

### CODE REVIEW

File: mumctl/src/main.rs [hunk 1]
```diff
@@ -223,6 +223,10 @@ impl fmt::Display for CliError {
 }
 
 fn main() {
+    if std::env::args().any(|s| s.as_str() == "--version" || s.as_str() == "-V") {
+        println!("mumctl {}", env!("VERSION"));
+        return;
+    }
```
Comment (default-username-852) [inline mumctl/src/main.rs:229]: `clap` handles this for us.

**Review by default-username-852 — COMMENTED** [review]
The changes on mumd looks great, but I'm not sure that we want to roll with our own version handling instead of using the built in handling in `clap`.

File: mumctl/src/main.rs (hunk 1 above)
Comment (sornas) [reply mumctl/src/main.rs:229]: Not with our env-variable. This is the current behaviour.

```
$ cargo run --bin mumd -- --version
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
     Running `target/debug/mumd --version`
mumd v0.3-291-gd7a1738
$ cargo run --bin mumctl -- --version
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
     Running `target/debug/mumctl --version`
mumctl 0.4.0
```

Comment (sornas) [reply mumctl/src/main.rs:229]: We have build scripts in both mumd and mumctl that use `git-describe` so the version number contains the hash.

Comment (default-username-852) [reply mumctl/src/main.rs:229]: Maybe make which version you use toggleable behind a config flag? I agree that it's great to have the commit hash in development builds, but I don't think that that is something that is necessary for a final user. Maybe include the hash in debug builds but don't in release? Also, I think it's weird that mumd reports its version as `0.3` while mumctl reports its version as `0.4`

Comment (sornas) [reply mumctl/src/main.rs:229]: > Maybe make which version you use toggleable behind a config flag? I agree that it's great to have the commit hash in development builds, but I don't think that that is something that is necessary for a final user.

Yes, that works. I'll hook something up.

> Also, I think it's weird that mumd reports its version as 0.3 while mumctl reports its version as 0.4.

That's what this PR fixes. With these changes both mumd and mumctl report the same version with the hash.

Comment (default-username-852) [reply mumctl/src/main.rs:229]: > That's what this PR fixes. With these changes both mumd and mumctl report the same version with the hash.

:+1:

**Comment by sornas** [comment]
How about this behaviour?

```
$ cargo run --bin mumd -- --version
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
     Running `target/debug/mumd --version`
mumd v0.3-289-g6d9863c

$ MUM_VERSION="0.4.0-rc.1" cargo run --bin mumd -- --version
   Compiling mumd v0.4.0 (/home/gustav/dev/mum/mumd)
    Finished dev [unoptimized + debuginfo] target(s) in 6.25s
     Running `target/debug/mumd --version`
mumd 0.4.0-rc.1

$ cargo run --bin mumd -- --version
   Compiling mumd v0.4.0 (/home/gustav/dev/mum/mumd)
    Finished dev [unoptimized + debuginfo] target(s) in 5.97s
     Running `target/debug/mumd --version`
mumd v0.3-289-g6d9863c
```

**Comment by default-username-852** [comment]
That seems fine

**Comment by sornas** [comment]
Pushed

**Review by default-username-852 — APPROVED** [review]
Looks good! Just remember to manually set the version when building prebuilt binaries for releases.

**Pull request merged**